fhevm/tee_interpreter.go

```diff
@@ -0,0 +1,181 @@
+package fhevm
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/zama-ai/fhevm-go/fhevm/tfhe"
+	"github.com/zama-ai/fhevm-go/tee"
+	"go.opentelemetry.io/otel/trace"
+)
+
+func doOperationGeneric[T any](op string, environment EVMEnvironment, caller common.Address, input []byte, runSpan trace.Span, operator func(uint64, uint64) T) ([]byte, error) {
```

Comment:
Would recommend putting one argument per line.

Also, small nit, let's put the `op` arg position last or before last, next to the operator argument